Golem/Yagna/Job.cs

```diff
@@ -101,6 +100,24 @@ public List<Payment> PaymentConfirmation
             }
         }
 
+        private DateTime? _timestamp = new DateTime();
+        public DateTime? Timestamp
+        {
+            get
+            {
+                return _timestamp;
+
+            }
+            set
+            {
+                if (_timestamp != value)
+                {
+                    _timestamp = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
```

Comment:
This property should never change and it will be initialized once at the start anyway
```suggestion
        public DateTime Timestamp { get; init; }
```